src/server.py

```diff
@@ -146,42 +140,31 @@ def assemble_response(request):
 
 
 def manage_client(request, conn):
+    """Handle incoming client requests and response messages."""
     try:
         response_msg = assemble_response(request)
     except RequestError as ex:
         response_msg = response_error(*ex.args)
     except IOError:
         response_msg = response_error(404, "File Not Found")
+    except OSError:
+        response_msg = response_error(404, "File Not Found")
```

Comment:
I was wondering about the errors you are catching here,then I went and looked at your `RequestError` class.  Very good!